src/FakeItEasy/Core/ArgumentValueFormatter.cs

```diff
@@ -25,6 +25,7 @@ public ArgumentValueFormatter(IEnumerable<IArgumentValueFormatter> typeFormatter
                     });
         }
 
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Any type of exception may be encountered.")]

```

Comment:
I don't think this warning makes sense when there's a `when` clause anyway... This should probably be changed in FxCop analyzers.